tensorflow/python/tools/optimize_for_inference_lib.py

```diff
@@ -531,3 +534,246 @@ def fuse_resize_and_conv(input_graph_def, output_node_names):
 
   result_graph_def.node.extend(new_ops)
   return result_graph_def
+
+def fuse_decomposed_batch_norm(input_graph_def):
+  """Fuse individual ops in batch normalization to FusedBatchNorm.
+
+  In some models, the batch normalizatin is performed via a group of individual
+  ops instead of using single FusedBatchNorm op. This function identifies a
+  pattern of batch normalization subgraph which is made of multiple ops and
+  transforms the graph by replacing those individual ops with FusedBatchNorm op.
+  This will provide the opportunity to further fold the FusedBatchNorm with
+  convolution ops to reduce the computation steps during inference.
+  This function currently only recognizes batch normalization patterns described
+  below, this could be extended if newer patterns are seen. Also, only target
+  for NHWC formatted graph.
+
+  Computation function:
+    (X * multiplier) + (Beta - Mean * multiplier)
+      where multiplier = rsqrt (Variance + Epsilon) * Gamma
+                    OR = rsqrt (Variance + Epsilon) when Gamma is 1
```

Comment:
Just curious, do you know where this specific form comes from? Does it come from a graph generated from TF or from somewhere else? Asking just because it's more natural to write batch norm this way: 
```
(X - Mean) * multiplier + Beta
```

Reply:
The patterns are seen in the densenet(-169), googlenet-v3 frozen pb files. I consulted with my teamates on this question, this pattern is produced as a call to tf.nn.batch_normalization in earlier TF version.